src/authentication.ts

```diff
@@ -54,14 +55,45 @@ export async function redirectToLogin(
     if (config?.preLogin) config.preLogin()
 
     if (method === 'popup') {
-      const handle: null | WindowProxy = window.open(loginUrl, 'loginPopup', 'popup width=600 height=600')
+      const { left, top, width, height } = calculatePopupPosition(600, 600)
+      const handle: null | WindowProxy = window.open(
+        loginUrl,
+        'loginPopup',
+        `width=${width},height=${height},top=${top},left=${left}`
+      )
       if (handle) return
       console.warn('Popup blocked. Redirecting to login page. Disable popup blocker to use popup login.')
     }
     window.location.assign(loginUrl)
   })
 }
 
+function calculatePopupPosition(popupWidth = 600, popupHeight = 600): TPopupPosition {
+  // Calculate the screen dimensions and position the popup at the center
+  const screenLeft = window.screenLeft || window.screenX
+  const screenTop = window.screenTop || window.screenY
+  const screenWidth = window.innerWidth || document.documentElement.clientWidth || screen.width
+  const screenHeight = window.innerHeight || document.documentElement.clientHeight || screen.height
```

Comment:
Will any of these actually be unset at any point? 

E.g. `screenLeft` vs `screenX` has basically the same stats on caniuse ([screenLeft](https://caniuse.com/mdn-api_window_screenleft), [screenX](https://caniuse.com/mdn-api_window_screenx)), and I assume the same may be true for the other 6 properties you've listed.

If so, let's just use the most "natural"/simple ones, and stick with those.

Reply:
And as such, if these are set in (basically) all browsers, consider using them directly further down into the script.